From 3cb41850fa8b2159e9a48a9298d5d3f0e114cbf4 Mon Sep 17 00:00:00 2001
From: Winny <winston@ml1.net>
Date: Tue, 12 Nov 2019 03:19:19 -0600
Subject: [PATCH 2/3] Separate GCC and GNU Make detection (#2902)

Related to #2890.

This ensures when GNU Make is `make`, the `version.mak` file is used
regardless of `CC`.
---
 racket/src/cfg-racket          | 9 +++++++--
 racket/src/racket/configure.ac | 9 +++++++--
 2 files changed, 14 insertions(+), 4 deletions(-)

diff --git a/racket/src/cfg-racket b/racket/src/cfg-racket
index a075ad87d0..abd2b2710e 100755
--- a/racket/src/cfg-racket
+++ b/racket/src/cfg-racket
@@ -4607,23 +4607,28 @@ if test "$ARFLAGS" = '' ; then
 fi
 SUB_CONFIGURE_EXTRAS="${SUB_CONFIGURE_EXTRAS} AR_FLAGS="'"'"${ARFLAGS}"'"'
 
+is_gmake=`make -v no-such-target-we-hope 2>&1 | grep "GNU Make"`
+
 # If using gcc, we want all warnings:
 if test "$CC" = "gcc" ; then
   COMPFLAGS="$COMPFLAGS -Wall"
 
   # Use -MMD when we have gcc and gnumake:
-  is_gmake=`make -v no-such-target-we-hope 2>&1 | grep "GNU Make"`
   if test "$is_gmake" = "" ; then
     WX_MMD_FLAG=""
   else
     WX_MMD_FLAG="-MMD"
-    INCLUDEDEP="-include"
   fi
 
   # Compile mzdyn.o with -fPIC
   DYN_CFLAGS="-fPIC"
 fi
 
+# When GNU Make is present, use version.mak
+if test "$is_gmake" != ""; then
+   INCLUDEDEP="-include"
+fi
+
 if test "$REZ" = '' ; then
  REZ=/Developer/Tools/Rez
 fi
diff --git a/racket/src/racket/configure.ac b/racket/src/racket/configure.ac
index 98c897c468..c563aa4409 100644
--- a/racket/src/racket/configure.ac
+++ b/racket/src/racket/configure.ac
@@ -460,23 +460,28 @@ if test "$ARFLAGS" = '' ; then
 fi
 SUB_CONFIGURE_EXTRAS="${SUB_CONFIGURE_EXTRAS} AR_FLAGS="'"'"${ARFLAGS}"'"'
 
+is_gmake=`make -v no-such-target-we-hope 2>&1 | grep "GNU Make"`
+
 # If using gcc, we want all warnings:
 if test "$CC" = "gcc" ; then 
   COMPFLAGS="$COMPFLAGS -Wall"
 
   # Use -MMD when we have gcc and gnumake:
-  is_gmake=`make -v no-such-target-we-hope 2>&1 | grep "GNU Make"`
   if test "$is_gmake" = "" ; then
     WX_MMD_FLAG=""
   else
     WX_MMD_FLAG="-MMD"
-    INCLUDEDEP="-include"
   fi
 
   # Compile mzdyn.o with -fPIC
   DYN_CFLAGS="-fPIC"
 fi
 
+# When GNU Make is present, use version.mak
+if test "$is_gmake" != ""; then
+   INCLUDEDEP="-include"
+fi
+
 if test "$REZ" = '' ; then 
  REZ=/Developer/Tools/Rez
 fi
-- 
2.25.0

